lms/security.py

```diff
@@ -48,9 +48,51 @@ def get_policy(request):
         return LMSGoogleSecurityPolicy()
 
     if path in {"/lti_launches", "/content_item_selection", "/api/gateway/h/lti"}:
+        # Actual LTI backed authentication
         return LTIUserSecurityPolicy(get_lti_user_from_launch_params)
 
-    return LTIUserSecurityPolicy(get_lti_user)
+    if path.startswith("/api") and path.endswith("callback"):
+        # LTIUser serialized in the state param for the oauth flow
+        return LTIUserSecurityPolicy(get_lti_user_from_oauth_callback)
```

Comment:
Sent as the state param here:

https://github.com/hypothesis/lms/blob/main/lms/views/api/d2l/authorize.py#L20 and similar routes.